### PR TITLE
Allow arbitrary options to be passed to the extension for rendering alternative sms providers

### PR DESCRIPTION
**Pull request by pavinduLakshan**
### Purpose

$subject

### Related Issues
- N/A

### Related PRs
- N/A

### Checklist
- [ ] e2e cypress tests locally verified.
- [X] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Integration tests provided. (Add links if there are any)

### Security checks
- [X] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [X] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: features/admin.extensions.v1/configs/models/sms-provider.ts [hunk 1]
```diff
@@ -19,7 +19,7 @@
 import { ReactNode } from "react";
 
 interface SmsProviderConfig {
-    renderAlternativeSmsProviderOptions: () => ReactNode
+    renderAlternativeSmsProviderOptions: (options: any) => ReactNode
```
Comment (DonOmalVindula) [inline features/admin.extensions.v1/configs/models/sms-provider.ts:22]: Is it possible to avoid using `any` here?

**Review by DonOmalVindula — APPROVED** [review]

File: features/admin.extensions.v1/configs/models/sms-provider.ts (hunk 1 above)
Comment (pavinduLakshan) [reply features/admin.extensions.v1/configs/models/sms-provider.ts]: Fixed with [c250bad](https://github.com/wso2/identity-apps/pull/6080/commits/c250badf35df570629e049a5fdbe36eaa67b3b58)

File: pnpm-lock.yaml
Comment (JayaShakthi97) [inline pnpm-lock.yaml:1]: Are these changes expected?

Comment (pavinduLakshan) [reply pnpm-lock.yaml:1]: Nope. removed

**Review by DonOmalVindula — APPROVED** [review]

**Pull request merged**